File: src/main/java/com/keycloak/reactive/security/JwtAuthConverterProperties.java
package com.keycloak.reactive.security;

import lombok.Getter;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.boot.context.properties.bind.ConstructorBinding;

@Getter
@ConfigurationProperties(prefix = "jwt.auth.converter")
public class JwtAuthConverterProperties {
    private final String resourceId;
    private final String principalAttribute;

    @ConstructorBinding
    public JwtAuthConverterProperties(
            String resourceId,
            String principalAttribute
    ) {
        this.resourceId = resourceId;
        this.principalAttribute = principalAttribute;
    }
}

File: src/main/kotlin/com/keycloak/reactive/security/JwtAuthConverterProperties.kt
package com.keycloak.reactive.security

import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.boot.context.properties.bind.ConstructorBinding

@ConfigurationProperties(prefix = "jwt.auth.converter")
data class JwtAuthConverterProperties @ConstructorBinding constructor(
    val resourceId: String,
    val principalAttribute: String?
)
